packages/expo-ui/components/ListItem/index.tsx

```diff
@@ -0,0 +1,124 @@
+   */
+  leadingActions?: SwipeAction;
+
+  /**
+   * Whether to activate the first action when user overshoots swipe from leading edge
+   * @default true
+   */
+  allowsFullSwipeLeading?: boolean;
+
+  /**
+   * Actions which will appear when swiping from the trailing edge
+   */
+  trailingActions?: SwipeAction;
+
+  /**
+   * Whether to activate the first action when user overshoots swipe from trailing edge
+   * @default true
+   */
+  allowsFullSwipeTrailing?: boolean;
+
+  /**
+   * Whether to hide trailing edge actions when list edit mode is active
+   * @default true
+   */
+  hideTrailingActionsInEditMode?: boolean;
+
+  children: React.ReactNode;
+};
+
+type ListItemNativeProps = Omit<ListItemProps, 'leadingActions' | 'trailingActions'> & {
+  leadingActions?: NativeSwipeActionProps[];
+  trailingActions?: NativeSwipeActionProps[];
+  onActionPressed: (event: SwipeActionNativeEvent) => void;
+};
+
+const ListItemNativeView: React.ComponentType<ListItemNativeProps> = requireNativeView(
+  'ExpoUI',
+  'ListItemView'
+);
+
+/**
+ * Provides a way to add swipe actions to List items
+ *
+ * @param {ListItemProps} props - The properties passed to the item.
+ * @returns {JSX.Element} The rendered native component.
+ * @platform ios
+ */
+export function ListItem({ leadingActions, trailingActions, children, ...props }: ListItemProps) {
+  const eventHandlersMap = new Map<string, (event: NativeSyntheticEvent<any>) => void>();
+
+  const processedLeadingActions = useMemo(
+    () => transformSwipeActions(leadingActions, eventHandlersMap),
+    [leadingActions]
+  );
+  const processedTrailingActions = useMemo(
+    () => transformSwipeActions(trailingActions, eventHandlersMap),
+    [trailingActions]
+  );
+
+  return (
+    <ListItemNativeView
+      {...props}
+      leadingActions={processedLeadingActions}
+      trailingActions={processedTrailingActions}
+      onActionPressed={(ev) => eventHandlersMap.get(ev.nativeEvent.id)?.call(undefined, ev)}>
+      {children}
+    </ListItemNativeView>
+  );
+}
+
+function isButtonElement(element: ReactElement<any>): element is SwipeActionElement {
+  return element.type === Button;
+}
+
+function processSwipeAction(
+  child: ReactNode,
+  eventHandlersMap: EventHandlerMap
+): NativeSwipeActionProps | null {
+  const isValidChild = isValidElement(child);
+  if (!isValidChild || !isButtonElement(child)) {
+    console.warn('Unsupported button type in ListItem ', isValidChild ? child?.type : child);
+    return null;
+  }
+
+  const uuid = expo.uuidv4();
+
+  if (child.props.onPress) eventHandlersMap.set(uuid, child.props.onPress);
+
+  return {
+    id: uuid,
+    props: transformButtonProps(child.props),
+  };
+}
+
+function getChildren(node?: SwipeAction) {
+  if (isValidElement(node) && node.type === Fragment) return Children.toArray(node.props.children);
+
+  return Children.toArray(node);
+}
+
+export function transformSwipeActions(
```

Comment:
This is inspired by `<ContextMenu />` maybe it would make sense to refactor it into a generic helper

```ts
transformChildren<TChild extends ReactElement>(
  children: TChild | React.ReactElement<{ children: TChild[] }>,
  eventHandlerMap: Map<string, (event: SwipeActionNativeEvent) => void>,
  transformer: ((childProps: React.ComponentProps<TChild>) => unknown)
)
```

and make it the default way to pass children

Original source:

https://github.com/expo/expo/blob/29090e430685255f311462fa9d355386e0b8326b/packages/expo-ui/components/ContextMenu/utils.ts#L46